modules/post/windows/gather/credentials/whatsupgold_credential_dump.rb

```diff
@@ -0,0 +1,651 @@
+##
+# This module requires Metasploit: https://metasploit.com/download
+# Current source: https://github.com/rapid7/metasploit-framework
+##
+
+require 'metasploit/framework/credential_collection'
+
+class MetasploitModule < Msf::Post
+  include Msf::Post::Common
+  include Msf::Post::File
+  include Msf::Post::Windows::MSSQL
+  include Msf::Post::Windows::Powershell
+  include Msf::Post::Windows::Registry
+
+  Rank = ManualRanking
+  def initialize(info = {})
+    super(
+      update_info(
+        info,
+        'Name' => 'WhatsUp Gold Credentials Dump',
+        'Description' => %q{
+          This module exports and decrypts credentials from WhatsUp Gold to a CSV file;
+          it is intended as a post-exploitation module for Windows hosts with WhatsUp
+          Gold installed. The module has been tested on and can successfully decrypt
+          credentials from WhatsUp versions 11.0 to the latest (22.x). Extracted
+          credentials are automatically added to loot.
+        },
+        'Author' => [
+
+          'sshah[at]assetnote.io', # original research
+          'npm[at]cesium137.io' # additional research and module
+        ],
+        'Platform' => [ 'win' ],
+        'DisclosureDate' => '2022-11-22',
+        'SessionTypes' => [ 'meterpreter' ],
+        'License' => MSF_LICENSE,
+        'References' => [
+          ['URL', 'https://blog.assetnote.io/2022/06/09/whatsup-gold-exploit/']
+        ],
+        'Actions' => [
+          [
+            'Dump',
+            {
+              'Description' => 'Export WhatsUp Gold database and perform decryption'
+            }
+          ],
+          [
+            'Export',
+            {
+              'Description' => 'Export WhatsUp Gold database without decryption'
+            }
+          ],
+          [
+            'Decrypt',
+            {
+              'Description' => 'Decrypt WhatsUp Gold database export CSV file'
+            }
+          ]
+        ],
+        'DefaultAction' => 'Dump',
+        'Notes' => {
+          'Stability' => [ CRASH_SAFE ],
+          'Reliability' => [ REPEATABLE_SESSION ],
+          'SideEffects' => [ IOC_IN_LOGS ]
+        },
+        'Privileged' => true
+      )
+    )
```

Comment:
Add CVEs?
From linked blog:
[CVE-2022-29845](https://nvd.nist.gov/vuln/detail/CVE-2022-29845): Local File Disclosure
[CVE-2022-29846](https://nvd.nist.gov/vuln/detail/CVE-2022-29846): WhatsUp Gold Serial Number Disclosure
[CVE-2022-29847](https://nvd.nist.gov/vuln/detail/CVE-2022-29847): Unauthenticated Server-Side Request Forgery (SSRF)
[CVE-2022-29848](https://nvd.nist.gov/vuln/detail/CVE-2022-29848): Authenticated Server-Side Request Forgery (SSRF)

Reply:
That's a good habit I need to develop. Done!